### PR TITLE
API dump

### PR DESCRIPTION
**Pull request by CherryPerry**
## Description

1. Disable generation of `BuildConfig` and etc. to remove them from API (consider as non-breaking, nobody uses them).
2. Generate and save public API dump files. Later these files will be used to verify API stability.
3. Provide guidance in PR template.

Sample from Reaktive repo: https://github.com/badoo/Reaktive/pull/684/files#diff-3072ccd590c25fcd425c2e6496a574be101b854d583830b6e34c4514ede81bd6R392

Part of https://github.com/bumble-tech/appyx/issues/177

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -6,3 +6,4 @@
 
 - [ ] I have updated `CHANGELOG.md` if required.
 - [ ] I have updated documentation if required.
+- [ ] I have regenerated API dump via `./gradlew apiDump` task.
```
Comment (LachlanMcKee) [inline .github/pull_request_template.md:9]: Perhaps as part of CI we could also call `apiDump` and compare it against the PR? We could then fail the build if there is a discrepancy.

That way we don't need this checkbox, and we will avoid the situation where someone regenerates it on their PR and several other PR changes are added into that PR, making reviewing confusing!

File: libraries/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+plugins {
```
Comment (LachlanMcKee) [inline libraries/build.gradle.kts:1]: interesting! didn't realise this works applies to all the child modules!

Comment (LachlanMcKee) [reply libraries/build.gradle.kts:1]: Does this then cause this to become a gradle module? (I'm unsure if it was already considered that way before)

File: .github/pull_request_template.md (hunk 1 above)
Comment (CherryPerry) [reply .github/pull_request_template.md:9]: `apiCheck` task is already part of `build` task chain. The problem that I faced in Reaktive, that people do not really check logs and errors, and most of the time we need to tell them explicitly to run this task.

File: libraries/build.gradle.kts (hunk 2 above)
Comment (CherryPerry) [reply libraries/build.gradle.kts:1]: `libraries` now is a Gradle module with no tasks but with a plugin.
The plugin is not applied to child modules automatically if it is declared in the parent. The plugin invokes `subprojects { }` internally to apply itself to all subprojects (`libraries:...)`.

File: .github/pull_request_template.md (hunk 1 above)
Comment (LachlanMcKee) [reply .github/pull_request_template.md:9]: They will have to check them if the CI prevents them from merging :D

File: libraries/build.gradle.kts (hunk 2 above)
Comment (LachlanMcKee) [reply libraries/build.gradle.kts:1]: thanks for the explanation!

**Comment by CherryPerry** [comment]
We do not want to support binary compatibility, only source one, because we do not expect the library to be used as a transitive dependency.

**Pull request closed without merging**